### ISSUE TITLE
 ld returned 1 exit status while go get

### ISSUE THREAD

**Issue by fscnick** [gographics/imagick#222]
Here is the error message:
```
# gopkg.in/gographics/imagick.v3/imagick
/tmp/go-build709501177/b134/_x003.o: In function `HeapOverflowSanityCheck':
/usr/include/ImageMagick-7/MagickCore/memory_.h:74: multiple definition of `HeapOverflowSanityCheck'
/tmp/go-build709501177/b134/_x002.o:/usr/include/ImageMagick-7/MagickCore/memory_.h:74: first defined here
/tmp/go-build709501177/b134/_x003.o: In function `HeapOverflowSanityCheckGetSize':
/usr/include/ImageMagick-7/MagickCore/memory_.h:86: multiple definition of `HeapOverflowSanityCheckGetSize'
/tmp/go-build709501177/b134/_x002.o:/usr/include/ImageMagick-7/MagickCore/memory_.h:86: first defined here
/tmp/go-build709501177/b134/_x004.o: In function `HeapOverflowSanityCheck':
/usr/include/ImageMagick-7/MagickCore/memory_.h:74: multiple definition of `HeapOverflowSanityCheck'
/tmp/go-build709501177/b134/_x002.o:/usr/include/ImageMagick-7/MagickCore/memory_.h:74: first defined here
/tmp/go-build709501177/b134/_x004.o: In function `HeapOverflowSanityCheckGetSize':
/usr/include/ImageMagick-7/MagickCore/memory_.h:86: multiple definition of `HeapOverflowSanityCheckGetSize'
...
...
...
/tmp/go-build709501177/b134/_x075.o: In function `HeapOverflowSanityCheckGetSize':
/usr/include/ImageMagick-7/MagickCore/memory_.h:86: multiple definition of `HeapOverflowSanityCheckGetSize'
/tmp/go-build709501177/b134/_x002.o:/usr/include/ImageMagick-7/MagickCore/memory_.h:86: first defined here
collect2: error: ld returned 1 exit status
```

This might be the similar issue: https://github.com/Imagick/imagick/issues/311

**Comment by justinfx** [comment]
Looks like they correlated it to a recent bug in ImageMagick. Are you building against a recent release that contains this bug? What version of ImageMagick?

**Comment by fscnick** [comment]
Thanks for replying. I installed via `remi`(Remi's RPM repository). The version of ImageMagick over there is `7.0.9.9-1`.They haven't catch up the latest fix.

It seems to need to wait for the update or use an older version.

**Comment by justinfx** [comment]
The CI tests for this binding are currently set to run against ImageMagick 6.9.10-11 and 7.0.8-11, which are over a year old now. So yes it seems because of their more recent bug you would need to build against an older version or a newer patch.
If you mean 7.0.9-9, they released that one 13 days ago. You must be using a package repo that provides very recent builds.

**Comment by justinfx** [comment]
I've just updated the master (im6) and im7 branches to use a docker container to run the tests, and I've build and pushed some new ImageMagick images to:

https://hub.docker.com/repository/docker/gographics/imagick.v2
https://hub.docker.com/repository/docker/gographics/imagick.v3

The travis-ci tests have been updated and are passing against both the latest ImageMagick7 and ImageMagick7 versions. So it looks like the bug has been fixed, but there are a range of versions where that bug does exist. So if you are using a package repo that provides one of those buggy versions, then you will have to find another package repo or build it from source for the desired version. Or use a docker container to provide the ImageMagick library.
